notebook/static/notebook/js/notebook.js

```diff
@@ -164,29 +164,79 @@ import {ShortcutEditor} from 'notebook/js/shortcuteditor';
         var that = this;
 
         Object.defineProperty(this, 'line_numbers', {
-          get: function(){
-            var d = that.config.data||{}
-            var cmc =  (d['Cell']||{})['cm_config']||{}
-            return cmc['lineNumbers'] || false;
-          },
-          set: function(value){
-            this.get_cells().map(function(c) {
-              c.code_mirror.setOption('lineNumbers', value);
-            })
-            that.config.update({'Cell':{'cm_config':{'lineNumbers':value}}})
-          }
-          
-        })
+            get: function() {
+                var d = that.config.data || {};
+                var cmc =  (d['Cell'] || {}) ['cm_config'] || {};
+                return cmc['lineNumbers'] || false;
+            },
+            set: function(value) {
+                this.get_cells().map(function(c) {
+                    c.code_mirror.setOption('lineNumbers', value);
+                });
+                that.config.update({
+                    'Cell': {
+                        'cm_config': {
+                            'lineNumbers':value
+                        }
+                    }
+                });
+            }
+        });
+        
+        Object.defineProperty(this, 'header', {
+            get: function() {
+                return that.class_config.get_sync('Header');
```

Comment:
I switched from ConfigSection to ConfigWithDefaults to take advantage of defaults.